Enterprise release 2024.10.0 (#288)
pganalyze/pganalyze-docs: enterprise/releases/2024-10-0.mdx

---
title: '2024.10.0 Release'
backlink_href: /docs/enterprise/releases
backlink_title: 'pganalyze Enterprise Server: Release Changelog'
---

**Docker tag:** `quay.io/pganalyze/enterprise:v2024.10.0`

## Notes

- This release requires a database migration when coming from older releases, be sure to follow the [recommended upgrade steps](/docs/enterprise/upgrade)
- **Important:** If you were previously running a version older than v2024.04.X we highly recommend first upgrading to v2024.04.5 and staying there for 7 to 35 days
  - The v2024.04.0 release introduced a backend change for how query statistics are stored. Starting that release the old format was still supported in a dual write mode.
  - Starting with this release the new packed format is required and the only one supported.

>   - If you have not been on v2024.04.X for sufficient time the migrations will fail with an error to prevent accidental loss of query statistics.

Reviewer: Should we note the optional override here? If not, users who read the release notes may assume there is no direct upgrade path (even if you're willing to lose that data). I think that's okay, just calling it out.

Author: I think its okay to not mention it since I'd rather not have people set that by accident, and the warning is directly telling you what to do.

- This release removes the use of the **Object storage**
  - With the new packed storage format, pganalyze Enterprise Server now stores all data within the main database, including log texts
  - This potentially affects database storage size, although the pganalyze team has actually observed storage reductions due to the packed format on our cloud-based environment, despite the extra data being stored
  - Additionally we recommend reviewing your encryption configuration for your database server, since pganalyze now relies on the database storage to be encrypted for security of log texts (instead of a separate log encryption key)

>   - After upgrading the installation will still be able to read from the object storage for old logs, but no longer write to it. After 7 days have passed (the log retention time) you can remove all object storage related configuration.

Author: Hmm, not sure if linking to the docs PR is actually useful, its a little bit hard to extract the settings from there. I'll leave as is for now.

## Features

- Index Advisor

>   - Add support for "pg_bigm", "cube" and "earthdistance" extensions <!-- 3953 -->

Reviewer: Based on the broken syntax highlighting, GitHub does not seem to like these HTML comments in markdown like this (so other markdown tools may also). But I guess if it doesn't seem to cause any problems, that's fine.

Author: Yeah, I've done this in previous releases, I think its fine despite the bad syntax highlighting.

  - Avoid errors when unsupported index types/data types are referenced by a table <!-- 3954 -->
  - Add support for "ltree" extension <!-- 3957 -->
  - Add ability to filter out acknowledged state insights <!--3943  -->
  - Fix the count in "QUERIES" column in insights list <!-- 4288 -->
    - In the Missing Indexes tab in the Index Advisor page, the column "QUERIES" was showing the number of missing indexes in that insight (table) previously. Now this column shows the number of unique affected queries due to these missing indexes as intended.
  - Show affected index count of each table in insights list <!-- 4289 -->
  - Recommend removing unused expression indexes in Unused Index alert <!-- 4185 -->
    - Previously expression indexes were excluded because they produce column statistics that the query planner can benefit from. But this was confusing for users who knew some indexes weren't being queried, and yet we didn't recommend deleting them. You may want to additionally `CREATE STATISTICS` for the expression when dropping an expression index if you notice worse query plans.
- Automated EXPLAIN and EXPLAIN Plans

>   - Automated EXPLAIN: Add a better support for handling new lines with the text log format <!-- 3937 -->

Reviewer: ```suggestion
  - Automated EXPLAIN: Add better support for handling newlines with the text log format <!-- 3937 -->
```

  - View EXPLAIN Source: Hide "Async Capable" field when converting from JSON to text format <!-- 3946 -->
  - Support all Aggregate nodes with text format explain <!-- 4020 -->
    - Previously, only the plain Aggregate with text format explain was supported. With this change, we support GroupAggregate, HashAggregate, and MixedAggregate better.
  - Revise EXPLAIN details layout <!-- 4174 -->
- Update bundled collector from 0.58.0 to 0.61.0 <!-- XXX -->

>   - See the [collector changelog](XXX) for the full details

Reviewer: ```suggestion
  - See the [collector changelog](https://github.com/pganalyze/collector/blob/main/CHANGELOG.md#0610------2024-10-23) for the full details
```

Author: Right, I wanted to update that and forgot, thanks :)

>   - See the [collector changelog](XXX) for the full details

Author: I missed that comment somehow - I think the memory allocation is a bit of a more complex situation with the bundled collector, since I think its actually other parts of the app that are more memory intensive (and I've not heard of any collector OOMs of the combined collector). I'll leave as is for now.

> - Raise connection tracing retention time from 24 hours to 7 days with new packed stats format

Author: Ah, good point - I'll revise to remove that last sentence then.

(I also noted in the same way in the newsletter which is already scheduled, but I think that small inaccuracy is okay there)

- Alert Policy: Enhanced Email notifications <!-- 4082 -->

>   - Email notifications are now sent by default for all critical severity issues. Previously, some critical issues did not trigger email notifications due to an oversight. This has been corrected.\nSeveral checks were missing from the Alert Policy page, which prevented email notifications for those checks. This update includes all checks on the Alert Policy page, making it possible to enable email notifications for every critical severity issue.

Reviewer: ```suggestion
  - Email notifications are now sent by default for all critical severity issues. Previously, some critical issues did not trigger email notifications due to an oversight. This has been corrected. Several checks were missing from the Alert Policy page, which prevented email notifications for those checks. This update includes all checks on the Alert Policy page, making it possible to enable email notifications for every critical severity issue.
```
Is this a literal "\n"? That seems like it should not be there.

Author: Err, yeah, that crept in from copying this from the changelog JSON file.

- Add more time predefined time periods to date picker <!-- 4125 -->
- Display a notice when a server is using an older collector <!-- 4162 -->
- Allow filtering Server and Database drop-down menus <!-- 4235 -->
  - This can be useful when working with a large number of servers or databases.
- System: Support showing system info and metrics for Azure databases <!-- 4241 -->
  - To enable this, set up the `AZURE_SUBSCRIPTION_ID` with the collector version 0.59.0 and above.

## Other improvements and bugfixes

- Query Details: EXPLAIN Plans tab now respects the date picker selection <!-- 4405 -->
  - Previously, EXPLAIN Plans tab in the query details page was showing only up to the most recent 50 EXPLAIN Plans, regardless of the time specified in the date picker.
  - With this change, it'll now respect the time selected in the date picker.

>   - In addition to this, introduce the pagination and also ability to search by the plan fingerprint to the list.

Reviewer: ```suggestion
  - In addition to this, introduce pagination and the ability to search by the plan fingerprint to the list.
```

- Remove compatibility layer for packed stats rollout <!-- 4050 -->
  - Release 2024.04.0 contained dual-write logic for internal stats tables to migrate to a new packed structure without requiring an expensive database migration or otherwise dropping historical data. This release drops the old table structure. As long as you have been running release 2024.04.x for 30 days, no data will be lost with this release.

> - Install Docs: Keep same generated password when moving install between steps <!-- 3933 -->

Reviewer: I don't think this applies to Enterprise Server, does it?

- Fix handling of single-line SQL comments in formatted queries <!-- 3979 -->
- Fix snapshot upload failures caused by file size limit <!-- 3986 -->
- Rename memory stats that belong to the OS to avoid confusion with Postgres `shared_buffers` <!-- 3988 -->
- Per-table VACUUM information: Fix rendering bug when autovacuum_vacuum_insert_threshold is 0 <!-- 4087 -->
- Schema Statistics: Increase width of schema name column in table listing <!-- 4183 -->
- Remove support for collector versions below 0.40.0 <!-- 4160 -->
- VACUUM Advisor: Hide negative bloat estimates <!-- 4219 -->
  - In some situations, bloat can be estimated as negative due to limitations of the bloat estimation heuristics. Hide estimates from these cases to avoid confusion.
- Improve date range selection on small screens <!-- 4229 -->
- EXPLAIN: Fix date range selector menu layout issue on smaller screens <!-- 4239 -->

> - Update a duplicate snapshot warning logic in Server Settings page <!-- 4240 -->

Reviewer: ```suggestion
- Update duplicate snapshot warning logic in Server Settings page <!-- 4240 -->
```

>   - Previously, this was wrongly providing a warning when the test snapshot is executed recently. This is now fixed.

Reviewer: ```suggestion
  - Previously, this was wrongly providing a warning when a test snapshot was executed recently. This is now fixed.
```

- System Stats: Fix CPU core count calculation <!-- 4244 -->

>   - We previously multiplied the CPU logical core count by the number of\nsockets to determine this number, but that's not the commonly expected\nway of calculating this. Also rename the field to "Logical Cores" to\nclarify the meaning.

Reviewer: ```suggestion
  - We previously multiplied the CPU logical core count by the number of sockets to determine this number, but that's not the commonly expected way of calculating this. Also rename the field to "Logical Cores" to clarify the meaning.
```

 LDAP Authentication: Require valid SSL certificate for LDAP server if secure connection is specified <!-- 4300 -->
- Schema Statistics: Show notice when the column stats are not available for the table due to outdated stats <!-- 4311 -->
  - When the statistics in pg_stats were out of sync with the current table structure, such as when columns have been added or removed since the last statistics collection, they were simply not shown previously. With this change, show notification so that ANALYZE can be run to solve the issue.
- EXPLAIN: Tweak plan fingerprinting to improve detection of partition indexes <!-- 4304 -->

>   - This will result in the same fingerprint for substantially identical\nquery plans that were previously fingerprinted distinctly.

Reviewer: ```suggestion
  - This will result in the same fingerprint for substantially identical query plans that were previously fingerprinted distinctly.
```

> - Update Heroku setup documentation <!-- 4333 -->

Reviewer: Also not relevant to ES, right?

- VACUUM Advisor: Fix loading vacuum run details when table information is not available <!-- 4360 -->
- Server: Fix link to AWS Console for RDS databases <!-- 4361 -->
- Schema Statistics: Show matching query scans on index detail page <!-- 4404 -->

>   - Now when viewing a specific index, you can see the matching table scans and associated queries.

Reviewer 2: It may be worth noting that this isn't supported for partitioned tables yet.
